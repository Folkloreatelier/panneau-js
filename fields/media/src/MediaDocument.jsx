import React from 'react';

import Media from './Media';

import classNames from './styles.scss';

const MediaDocument = props => (
    <Media
        {...props}
        className={classNames.document}
        getThumbnail={media => (
            typeof media.thumbnails !== 'undefined' && media.thumbnails.length ? media.thumbnails[0].url : null
        )}
        renderDetails={(media, common) => (
            <div>
                { common }
                { media.pages ? (
                    <div>Pages: { media.pages } page(s)</div>
                ) : null}
                { media.original_file ? (
                    <div>Poids: { media.original_file.size_human }</div>
                ) : null }
            </div>
        )}
    />
);

export default MediaDocument;
