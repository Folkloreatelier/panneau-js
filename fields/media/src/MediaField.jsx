import React, { Component } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { withDefinition, PropTypes as PanneauPropTypes } from '@panneau/core';
import { FormGroup, UploadButton } from '@panneau/field';

import messages from './messages';

import styles from './styles.scss';

const propTypes = {
    CardComponent: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
    name: PropTypes.string,
    label: PanneauPropTypes.label,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    type: PropTypes.string,
    cardVertical: PropTypes.bool,
    cardWithoutBorder: PropTypes.bool,
    onChange: PropTypes.func,
    uploadEndpoint: PropTypes.string,
    uploadButtonLabel: PanneauPropTypes.message,
};

const defaultProps = {
    name: null,
    label: null,
    value: null,
    cardVertical: false,
    cardWithoutBorder: false,
    onChange: null,
    type: null,
    uploadEndpoint: null,
    uploadButtonLabel: messages.uploadMediaLabel,
};

class MediaField extends Component {
    constructor(props) {
        super(props);

        this.onUploadComplete = this.onUploadComplete.bind(this);
        this.onClickDelete = this.onClickDelete.bind(this);
    }

    onClickDelete() {
        if (this.props.onChange) {
            this.props.onChange(null);
        }
    }

    onUploadComplete(response) {
        if (this.props.onChange) {
            this.props.onChange(response);
        }
    }

    renderUploadButton() {
        const { uploadButtonLabel, uploadEndpoint } = this.props;
        return (
            <div className={styles.upload}>
                <UploadButton
                    endpoint={uploadEndpoint}
                    label={uploadButtonLabel}
                    onUploadComplete={this.onUploadComplete}
                />
            </div>
        );
    }

    renderCard() {
        const {
            name,
            value,
            label,
            type,
            cardVertical,
            cardWithoutBorder,
            CardComponent,
            uploadButtonLabel,
            uploadEndpoint,
            ...cardProps
        } = this.props;
        return (
            <div className={styles.media}>
                {CardComponent !== null ? (
                    <CardComponent
                        vertical={cardVertical}
                        withoutBorder={cardWithoutBorder}
                        {...cardProps}
                        item={value}
                        onClickDelete={this.onClickDelete}
                    />
                ) : null}
            </div>
        );
    }

    render() {
        const {
            name,
            value,
            label,
            type,
            cardVertical,
            cardWithoutBorder,
            CardComponent,
            uploadButtonLabel,
            uploadEndpoint,
            ...formGroupProps
        } = this.props;

        return (
            <FormGroup className={styles.field} name={name} label={label} {...formGroupProps}>
                {value !== null ? this.renderCard() : this.renderUploadButton()}
            </FormGroup>
        );
    }
}

MediaField.propTypes = propTypes;
MediaField.defaultProps = defaultProps;

const mapDefinitionToProps = definition => ({
    uploadEndpoint: get(definition, 'endpointUploadMedia', '/mediatheque/upload'),
});
export default withDefinition(mapDefinitionToProps)(MediaField);
