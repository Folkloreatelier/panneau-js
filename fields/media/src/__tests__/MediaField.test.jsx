import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
/* eslint-enable import/no-extraneous-dependencies */
import { IntlProvider } from 'react-intl';
import MediaField from '../MediaField';
import MediaDocument from '../MediaDocument';

test('match snapshot', () => {
    const component = renderer.create((
        <IntlProvider locale="en">
            <MediaField
                label="Label"
                MediaComponent={MediaDocument}
            />
        </IntlProvider>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
