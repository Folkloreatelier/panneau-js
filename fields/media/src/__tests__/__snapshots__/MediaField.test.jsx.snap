// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="form-group text"
>
  <div
    className="form-group-label"
  >
    <label
      className="control-label"
      htmlFor={null}
    >
      Label
    </label>
  </div>
  <div
    className="form-group-inner"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div />
  </div>
</div>
`;
